perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3820,31 +3820,21 @@ private void storeApplicationAttributes(Application app) throws UserNotExistsExc
 				// if attribute exists
 				if (a != null) {
 					if (a.getType().equalsIgnoreCase(LinkedHashMap.class.getName())) {
-						// FIXME do not set hash map attributes - not supported in GUI and registrar
-						continue;
+
+						// we expect that map contains string keys and values
+						LinkedHashMap<String, String> value = a.valueAsMap();
+						value = handleMapValue(value, newValue);
+
+						a.setValue(value);
+						attributes.add(a);
 					} else if (a.getType().equalsIgnoreCase(ArrayList.class.getName())) {
 
 						// we expect that list contains strings
 						ArrayList<String> value = a.valueAsList();
+						value = handleArrayValue(value, newValue);
```

Comment:
I think this method would correctly handle only values ending with `,`.  We should still support current behavior, e.g. `TEXTFIELD` with ArrayList destination attribute - its value doesn't end with `,`. So if it is an ArrayList attribute, we should handle differently `LIST_INPUT_BOX` type (method `handleArrayValue`) and other types (current behavior - don't split value, simply merge it to current value).

Reply:
I have decided to just add a `,`  to values from different widgets and that way they can be processed the same way.